src/metabase/search/appdb/core.clj

```diff
@@ -182,3 +186,11 @@
     (search.index/maybe-create-pending!))
   (u/prog1 (populate-index! (if in-place? :search/updating :search/reindexing))
     (search.index/activate-table!)))
+
+(derive :event/setting-update ::settings-changed-event)
+
+(methodical/defmethod events/publish-event! ::settings-changed-event
+  [_topic event]
+  (when (and (= :site-locale (-> event :details :key)) (= :postgres (mdb/db-type)))
```

Comment:
We're reindexing here when someone changes the `site-locale` while the application is running, but:
- not if `search-language` changes - is that intentional?
- not if someone uses an env var to change the site locale - is that okay? (Do we reindex on startup?)

Reply:
We do check the index on startup, so that should handle the case of changing the site-locale via an env variable.

Search-language isn't exposed via the UI so I didn't bother watching for changes since it can only be set by an env variable right? So then should fall into the "if there is no index (because the lang changed)" then reindex it case

Reply:
That makes sense - I wasn't sure if maybe we'd thought about exposing it someday.